### PR TITLE
fix: premium buttons and entitlement iterator

### PR DESCRIPTION
**Pull request by tibue99**
## Summary
This PR fixes some issues with monetization.

1) Premium buttons do not work with a custom ID, I disabled the auto-generated custom_id for premium buttons
2) `bot.fetch_entitlements()` did not work when certain parameters were not set
3) The `exclude_ended` attribute threw an error, because yarl does not support boolean params. I don't know how to fix that, so I just set it to `None` for now. Perhaps @plun1331 can help

I have tested the changes with a monetized Discord bot.

Edit: This PR now renames `client.fetch_entitlements` to `client.entitlements`.

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by plun1331** [comment]
> The exclude_ended attribute threw an error, because yarl does not support boolean params. I don't know how to fix that, so I just set it to None for now. Perhaps @plun1331 can help

Casting it to a string in the http method should be good enough

File: discord/iterators.py [hunk 1]
```diff
@@ -1055,7 +1055,7 @@ async def fill_entitlements(self):
             user_id=self.user_id,
             guild_id=self.guild_id,
             sku_ids=self.sku_ids,
-            exclude_ended=self.exclude_ended,
+            exclude_ended=int(self.exclude_ended),
```
Comment (plun1331) [inline discord/iterators.py:1058]: I meant under the get_entitlements method in discord/http

**Review by NeloBlivion — CHANGES_REQUESTED** [review]
While we're here, the implementation of `fetch_entitlements` should be rethought:
- As it returns an Iterator, it should NOT be an `async` function (matching `history`, `audit_logs`, reaction `users` etc.)
- In that vein, to match said functions it should ideally be renamed to just `entitlements` to follow standard iterator usage: 
```py
async for e in client.entitlements():
   ...

entitlements = await user.entitlements().flatten()
```

**Comment by Lulalaby** [comment]
Timo for the win. Thank you

**Review by NeloBlivion — APPROVED** [review]

**Review by Dorukyum — CHANGES_REQUESTED** [review]
Changelog

File: discord/client.py [hunk 2]
```diff
@@ -2053,9 +2053,7 @@ async def fetch_entitlements(
         guild: Snowflake | None = None,
         exclude_ended: bool = False,
     ) -> EntitlementIterator:
-        """|coro|
-
-        Fetches the bot's entitlements.
+        """Returns an :class:`.AsyncIterator` that enables fetching the application's entitlements.
 
         .. versionadded:: 2.5
```
Comment (Dorukyum) [inline discord/client.py:2058]: This should technically say 2.6 now that it's been renamed.

Comment (NeloBlivion) [reply discord/client.py]: Ah... i was completely unaware that this was previously in 2.5, making this breaking. Should the original name perhaps be included as a deprecated function, or leave it as is?

**Comment by Dorukyum** [comment]
Have the latest changes been tested?

**Comment by tibue99** [comment]
> Have the latest changes been tested?

I have tested `client.entitlements()` in production, but I have not tested the user/guild methods separately.

**Review by Lulalaby — DISMISSED** [review]
LGTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -93,7 +93,10 @@ These changes are available on the `master` branch, but have not yet been releas
   `ApplicationCommand.contexts`.
   ([#2409](https://github.com/Pycord-Development/pycord/pull/2409))
 - `Message.interaction` is now deprecated in favor of `Message.interaction_metadata`.
-  ([#2409](https://github.com/Pycord-Development/pycord/pull/2409)
+  ([#2409](https://github.com/Pycord-Development/pycord/pull/2409))
+- Replaced `client.fetch_entitlements` with `client.entitlements`, which returns an
```
Comment (plun1331) [inline CHANGELOG.md:97]:
```suggestion
- Replaced `Client.fetch_entitlements` with `Client.entitlements`, which returns an
```

**Review by plun1331 — APPROVED** [review]

**Review by NeloBlivion — APPROVED** [review]

**Pull request merged**